python_modules/dagster/dagster/_core/definitions/declarative_scheduling/scheduling_context.py

```diff
@@ -30,82 +33,109 @@
-    # condition evaluation tree, this is the AllPartitionsSubset, but this may shrink as conditions
-    # are applied and we can be certain that certain partitions will not be true for the overall
-    # expression
-    candidate_subset: ValidAssetSubset
-
-    # a view of the AssetGraph that will be used to help compute properties of the asset during
-    # computation
     asset_graph_view: AssetGraphView
 
-    # the serialized information calculated during the previous evaluation of this condition
-    # note that this refers to the evaluation of this specific node in the condition tree, not the
-    # evaluation of the root of the tree
-    # used to avoid recomputing information that we know has not changed since the previous tick
-    previous_evaluation: Optional[AssetConditionEvaluation]
-
-    # contains information about the previous evaluation of all assets within this asset's automation
-    # policy sensor. this provides a pointer to the top-level condition in the evaluation tree, as
-    # well as a few extra fields, such as the previous evaluation timestamp, and the max event id
-    # at the time of that computation. this information is again used to avoid recomputing information
-    # as well as detecting if anything has changed since the previous time this condition was evaluated
-    previous_evaluation_state_by_key: Mapping[AssetKey, AssetConditionEvaluationState]
-    # this is the same as the above, but for information that has been calculated during this tick.
-    # it will be used to determine if a parent will be materialized on this tick
-    current_evaluation_state_by_key: Mapping[AssetKey, AssetConditionEvaluationState]
-
-    # the time at which this context object was created, allowing us to time the duration of an
-    # evaluation for display in the UI
+    # the context object for the parent condition
+    parent_context: Optional["SchedulingContext"]
+
+    # the time at which this context object was created
     create_time: datetime.datetime
     logger: logging.Logger
 
-    # we need to continue supporting the current implementations of AutoMaterializeRules,
-    # which rely on the legact context object. however, this object contains many fields
-    # which are not relevant to the scheduling condition evaluation, and so keeping it
-    # as a reference here makes it easy to remove it in the future.
-    #
-    # marked as Any to avoid circular imports in the pydantic validation logic
-    inner_legacy_context: Any  # AssetConditionEvaluationContext
+    # a SchedulingEvaluationInfo object representing information about the full evaluation tree
+    # from the previous tick, if this asset was evaluated on the previous tick
+    previous_evaluation_info: Optional[SchedulingEvaluationInfo]
+    # a mapping of information computed on the current tick for assets which are upstream of this
+    # asset
+    current_tick_evaluation_info_by_key: Mapping[AssetKey, SchedulingEvaluationInfo]
```

Comment:
Will we need a class the represents "all information about the tick so far"?

Reply:
resolving: answer is probably yes, but can cross that bridge in the future